src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -309,14 +304,33 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 throw new ArgumentNullException("stream");
             }
 
-            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>();
+            TaskCompletionSource<object> tcs = new TaskCompletionSource<object>(this);
             try
             {
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    _bufferedContent.Position = 0;
```

Comment:
This doesn't look correct. My previous comment was about remembering the position from bufferedContent, not forcing it to be 0.  The continuation would need to reset the position back to whatever it was prior to copying from it, right?  I don't know where this MemoryStream is coming from, so maybe this is moot and it's always at position 0 here, but I don't know.